src/modules/idyntree-yarp-visualizer/RobotConnectors.cpp

```diff
@@ -0,0 +1,558 @@
+            {
+                jointNamesVector.push_back(jointVector.back().name);
+            }
+        }
+
+        if (m_cb_jointsMap.back().jointsToConsider < 0)
+        {
+            m_cb_jointsMap.back().jointsToConsider = m_cb_jointsMap.back().joints.size();
+        }
+
+    }
+
+    {
+        std::lock_guard<std::mutex> lock(m_basicInfo->mutex);
+        m_basicInfo->jointList = jointNamesVector;
+    }
+
+    return true;
+}
+
+bool StateExtConnector::configure(const yarp::os::Searchable &inputConf, std::shared_ptr<BasicInfo> basicInfo)
+{
+    std::lock_guard<std::mutex> lock(m_mutex);
+
+    m_basicInfo = basicInfo;
+
+    yarp::os::Value confValue = inputConf.find("connectToStateExt");
+
+    if (confValue.isString() && confValue.asString() == "default")
+    {
+        std::string robotLocalName;
+        {
+            std::lock_guard<std::mutex> lock(m_basicInfo->mutex);
+
+            robotLocalName = m_basicInfo->robotPrefix;
+        }
+
+        if (robotLocalName == "icubSim")
+        {
+            confValue.fromString("(head, (neck_pitch, neck_roll, neck_yaw),"
+                                 " torso, (torso_yaw, torso_pitch, torso_roll),"
+                                 " left_arm, (l_shoulder_pitch, l_shoulder_roll, l_shoulder_yaw, l_elbow, l_wrist_prosup, l_wrist_pitch, l_wrist_yaw),"
+                                 " right_arm, (r_shoulder_pitch, r_shoulder_roll, r_shoulder_yaw, r_elbow, r_wrist_prosup, r_wrist_pitch, r_wrist_yaw),"
+                                 " left_leg, (l_hip_pitch, l_hip_roll, l_hip_yaw, l_knee, l_ankle_pitch, l_ankle_roll),"
+                                 " right_leg, (r_hip_pitch, r_hip_roll, r_hip_yaw, r_knee, r_ankle_pitch, r_ankle_roll))");
+        }
+        else if (robotLocalName == "icub")
+        {
+            confValue.fromString("((head, 3), (neck_pitch, neck_roll, neck_yaw, eyes_tilt, eyes_vers, eyes_verg),"
+                                 " torso, (torso_roll, torso_pitch, torso_yaw),"
+                                 " (left_arm, 7), (l_shoulder_pitch, l_shoulder_roll, l_shoulder_yaw, l_elbow, l_wrist_prosup, l_wrist_pitch, l_wrist_yaw,"
+                                    " l_hand_finger, l_thumb_oppose, l_thumb_proximal, l_thumb_distal, l_index_proximal, l_index_distal, l_middle_proximal, l_middle_distal, l_little_fingers),"
+                                 " (right_arm, 7), (r_shoulder_pitch, r_shoulder_roll, r_shoulder_yaw, r_elbow, r_wrist_prosup, r_wrist_pitch, r_wrist_yaw,"
+                                    " r_hand_finger, r_thumb_oppose, r_thumb_proximal, r_thumb_distal, r_index_proximal, r_index_distal, r_middle_proximal, r_middle_distal, r_little_fingers),"
+                                 " left_leg, (l_hip_pitch, l_hip_roll, l_hip_yaw, l_knee, l_ankle_pitch, l_ankle_roll),"
+                                 " right_leg, (r_hip_pitch, r_hip_roll, r_hip_yaw, r_knee, r_ankle_pitch, r_ankle_roll))");
+        }
```

Comment:
I totally agree with the idea "provide the default configuration for a generic fully fledged icub, and fallback to complex config file for other robots or non-standard iCubs. However, this specific logic may be confusing, as on Gazebo there are users that use `iCubGazeboV2_5_visuomanip` so they can access eyes and hands joints. Do you think it make sense to keep the default without eyes and hands also on the real robot?

Reply:
Not sure I understood the desiderata :grin: In principle, I could add another keyword, like visuoManip. The good point is that the name of the joints can be the one of the model. What would you propose?

Reply:
Not sure I fully understood your reply as well, better to talk briefly, I will try to contact you.

Reply:
We quickly discussed F2F and we agreed to keep it as it is.

Reply:
To recap a bit the rationale discuss in person: switching to just use the same set for icub and icubSim will not working out of the box for all supported iCub (in Gazebo) and on the real robot. Even if we supported the hands+eyes case as the default, there would further problems related to the fact that joint names of hands in URDF models and in yarprobotinterface do not match (see https://github.com/robotology/icub-models/issues/30). So, even if it is confusing that the default values depend on an unrelated parameter (the port prefix) let's keep as it is as it match the most common type of users (the one using the `iCubGazeboV2_5` on Gazebo), while other users can just define a configuration file.